codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -49,19 +50,15 @@ private HttpUtil() { }
      * <a href="https://tools.ietf.org/html/rfc7230#section-5.3">rfc7230, 5.3</a>.
      */
     public static boolean isOriginForm(URI uri) {
-        return uri.getScheme() == null && uri.getSchemeSpecificPart() == null &&
-               uri.getHost() == null && uri.getAuthority() == null;
+        return isNullOrEmpty(uri.getScheme()) && isNullOrEmpty(uri.getAuthority()) && uri.toString().startsWith("/");
```

Comment:
I don't think it is possible to have an empty scheme; only null. This check wouldn't be right for the path `//foo` which gets parsed as an authority even though it is an origin-form path.

I think it should just be `uri.toString().startsWith("/")`. These really need to be string comparisons and not deal with URI API.

Reply:
I'm assuming we don't want to delete methods from the public API, so I'll add an overload to each that takes a `String` (and redirect the `URI` methods to use those under the hood).

Reply:
I considered suggesting a String override, but I don't think it'd be used much. The URI here will generally be gotten from `httpRequest.uri()` and there's not a String version of that API, so the String version seems like it'd be unused.

Reply:
I'm confused. Do you mean https://netty.io/4.1/api/io/netty/handler/codec/http/HttpRequest.html? That *only* has a `String uri()` method, there's no `URI`-typed API there. In fact, providing the `String` overrides is the only reason I can use these methods in `HttpConversionUtil.java` now instead of manually doing the raw string checks.

Reply:
Yeah, that's what I was meaning. Huh, I looked at it wrong. String is good!